### DISCUSSION TITLE
ADB

### DISCUSSION THREAD

**Discussion by zovl1** [ntherning/StorytelReaderMods#4] [Q&A]
No matter how hard i try, i can't seem to get adb to work, any solutions? (Gen 1 reader)

**Comment by coffeetails** [comment]
I struggle with this too. adb just won't find the device. I have a gen 2 reader and use Ubuntu on my pc

**Reply by ntherning** [reply]
Please provide some more details. Did you manage to backup the boot partition and flash the `boot-gen2-patched.img` file successfully? What does `lsusb` output after you connect the patched device to your Ubuntu PC?

**Reply by coffeetails** [reply]
Yes, everything with the rkflashtool was successful. 

lsusb outputs:
```
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 002: ID 045e:00db Microsoft Corp. Natural Ergonomic Keyboard 4000 V1.0
Bus 001 Device 003: ID 045e:0810 Microsoft Corp. LifeCam HD-3000
Bus 001 Device 004: ID 2109:2813 VIA Labs, Inc. VL813 Hub
Bus 001 Device 005: ID 1038:1722 SteelSeries ApS SteelSeries Sensei 310 eSports Mouse
Bus 001 Device 006: ID 2109:2813 VIA Labs, Inc. VL813 Hub
Bus 001 Device 008: ID 0b0e:0300 GN Netcom Jabra EVOLVE 20 MS
Bus 001 Device 009: ID 056a:030e Wacom Co., Ltd CTL-480 [Intuos Pen (S)]
Bus 001 Device 018: ID 2207:310c Fuzhou Rockchip Electronics Company RK3126/RK3128 in Mask ROM mode
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 002 Device 002: ID 2109:0813 VIA Labs, Inc. VL813 Hub
Bus 002 Device 003: ID 2109:0813 VIA Labs, Inc. VL813 Hub
```

**Reply by coffeetails** [reply]
I found the problem now! After doing the patch of the boot image I never rebooted the device as it wasn't mention in the long instructions. But you did mention it in the short instructions at the beginning of the README file

**Reply by ntherning** [reply]
It looks like you're device is still in maskrom mode. You need to restart it into normal mode for ADB to work. Here's what you should see from `lsusb` when your device is in normal mode (this is on macOS but the ID should be the same on Linux):

`
Bus 000 Device 004: ID 2207:0011 Fuzhou Rockchip Electronics Co., Ltd. Reader 2  Serial: 0123456789ABCDEF
`

My gen 2 device will boot into normal mode when I run `rkflashtool b` while it is in maskrom mode. Did you try that? You should not press the orange button or anything when you issue this command.

**Reply by ntherning** [reply]
Aha, I am pleased to hear that you worked it out! I'll see if I can make the instructions clearer. Thanks!

**Reply by zovl1** [reply]
Sorry for not replying, i've been on vacation. I actually found a amazon kindle bw3 in my electric trash were i live, so haven't had the need to convert the reader anymore, thanks for the replies and help anyway!
